factory/heartbeatV2Components.go

```diff
@@ -87,6 +88,19 @@ func checkHeartbeatV2FactoryArgs(args ArgHeartbeatV2ComponentsFactory) error {
 
 // Create creates the heartbeatV2 components
 func (hcf *heartbeatV2ComponentsFactory) Create() (*heartbeatV2Components, error) {
+	if !hcf.networkComponents.NetworkMessenger().HasTopic(common.PeerAuthenticationTopic) {
```

Comment:
We might think of this when we will refactor the network messenger:
This pattern is widely used: if !hasTopic.... createTopic. We can have something like: createTopicIfMissing